pints/_mcmc/_slice_doubling.py

```diff
@@ -0,0 +1,583 @@
+                        self._current_log_y >= self._fx_r_hat):
+
+                    # Shrink the interval ``I=(l,r)``
+                    if (self._proposed[self._active_param_index] <
+                            self._current[self._active_param_index]):
+                        self._l = self._proposed[self._active_param_index]
+                        self._temp_l[self._active_param_index] = self._l
+                    else:
+                        self._r = self._proposed[self._active_param_index]
+                        self._temp_r[self._active_param_index] = self._r
+
+                    # Reset variables
+                    self._continue_check = False
+                    self._d = False
+                    self._fx_r_hat = None
+                    self._fx_l_hat = None
+
+                    return None
+
+                # If the rejection condition is not met, we continue the
+                # ``Acceptance Check`` while (r_hat - l_hat ) > 1.1*w
+                return None
+
+            # We have updated successfully the parameter of the new sample!
+
+            self._first_expansion = True
+            self._interval_found = False
+
+            # Reset active parameter index
+            if self._active_param_index == len(self._proposed) - 1:
+
+                self._active_param_index = 0
+
+                # The accepted sample becomes the new current sample
+                self._current = np.array(self._proposed, copy=True)
+
+                # The log_pdf of the accepted sample is used to construct the
+                # new slice
+                self._current_log_pdf = fx
+
+                # Sample new log_y used to define the next slice
+                self._e = np.random.exponential(1)
+                self._current_log_y = self._current_log_pdf - self._e
+
+                # Return the accepted sample
+                return np.array(self._proposed, copy=True)
+
+            else:
+                self._active_param_index += 1
+                return None
+
+        # If the trial point is rejected in the ``Threshold Check``, shrink
+        # the interval
+        if (self._proposed[self._active_param_index] <
+                self._current[self._active_param_index]):
+            self._l = self._proposed[self._active_param_index]
+            self._temp_l[self._active_param_index] = self._l
+        else:
+            self._r = self._proposed[self._active_param_index]
+            self._temp_r[self._active_param_index] = self._r
+
+        # If the point has been rejected, set flag for ``Acceptance Check``
+        # check to False so that we can sample a new point
+        self._init_check = False
+        self._continue_check = False
+
+        return None
+
+    def name(self):
+        """ See :meth:`pints.MCMCSampler.name()`. """
+        return 'Slice Sampling - Doubling'
+
+    def set_w(self, w):
+        """
+        Sets width w for generating the interval.
```

Comment:
What should the width be? A single number or an array? If an array, then what shape?